appveyor.yml

```diff
@@ -35,43 +37,21 @@ install:
   - ps: .\build\appveyor_install.ps1
   - ps: .\build\scassandra_install.ps1
 
-before_build:
-  - ps: dotnet restore
-
-build:
-  project: src/Cassandra.sln
-  verbosity: minimal
-
-test:
-  assemblies:
-    - src/Cassandra.Tests/bin/$(configuration)/Cassandra.Tests.dll
-    - src/Cassandra.IntegrationTests/bin/$(configuration)/Cassandra.IntegrationTests.dll
-
-  categories:
-    except:
-      - long
-      - duration
+build_script:
+  - ps: dotnet restore src
+  - ps: dotnet build src\Cassandra.sln -c Release
 
 test_script:
-  - ps: >-
-      If ("$($env:CI_TYPE)" -eq "UNIT") {
-        dotnet test src\Cassandra.Tests -c Release -f $env:TARGET --result=TestResult.xml
-      }
-      If ("$($env:CI_TYPE)" -eq "INTEGRATION") {
-        dotnet test src\Cassandra.IntegrationTests -c Release -f $env:TARGET --where "cat=short" --result=TestResult.xml
+  - ps: |
+      If ("$($env:CI_TYPE)" -ne "MEMORY") {
+        dotnet test src\${env:PROJECT}\${env:PROJECT}.csproj -c Release -f $env:TARGET --filter "(TestCategory!=short)&(TestCategory!=memory)"  --logger "trx;LogFileName=..\..\..\TestResult.xml"
       }
-      If ("$($env:CI_TYPE)" -eq "MEMORY") {
+      Else {
         .\build\memory_tests.ps1
       }
-
-after_test:
-  - ps: >-
-      if($true)
-      {
-        $wc = New-Object 'System.Net.WebClient'
-        $wc.UploadFile("https://ci.appveyor.com/api/testresults/nunit3/$($env:APPVEYOR_JOB_ID)", (Resolve-Path .\TestResult.xml))
-      }
-
+      xml tr tools\trx-to-junit.xslt TestResult.xml > ResultsJUnit.xml
```

Comment:
This should be moved to the section `on_finish`.
On appveyor, the build is shown as successful or failed depending on the exit code of the `ps` script.

TLDR: The `ps` script is evaluated block by block and when the last statement is executed, the exit code for the script is returned. In this case, the last statement would be `$wc.UploadFile()`: as it succeeds, the build is shown as successful, even though `dotnet test` exited with code != 0. See https://ci.appveyor.com/project/DataStax/csharp-driver/build/master-1166/job/510r3hd6flkxh26i for example.